Assets/Scripts/TowerSystem/NewTowerBase.cs

```diff
@@ -82,8 +83,14 @@ protected virtual void TryFire()
         protected virtual void OnFire()
         {
             var spawnPos = BulletSource.position + new Vector3(0, 1, 0);
-            var bullet = Instantiate(Config.BulletConfig.prefab, spawnPos, transform.rotation);
-            bullet.GetComponent<NewBulletBehaviour>().Initialize(Config.debuffs, TargetedEnemy, Config.BulletConfig);
+            var bullet = PoolManager.Instance.GetBullet(Config.BulletConfig.ID);
+            if (bullet != null)
+            {
+                bullet.Initialize(Config.debuffs, TargetedEnemy, Config.BulletConfig, spawnPos);
+            }
+            
+            // var bullet = Instantiate(Config.BulletConfig.prefab, spawnPos, transform.rotation);
```

Comment:
Can probably remove the old code here, we can always go back in git if anything :)